projects/backend/capi/articles.ts

```diff
@@ -22,64 +24,96 @@ interface CAPIArticle {
-    const body = blocks && blocks.reduce((acc, cur) => [...acc, ...cur], [])
-    if (body == null) throw new Error(`Body was undefined in ${path}!`)
-
-    const elements = await attempt(Promise.all(body.map(parser)))
-    if (hasFailed(elements))
-        throw new Error(`Element parsing failed in ${path}!`) //This should not fire, the parser should log if anything async fails and then return the remainder.
-
-    if (elements == null) throw new Error(`Elements was undefined in ${path}!`)
+    if (standfirst == null) console.error(`Standfirst was undefined in ${path}!`)
 
     const byline = result && result.fields && result.fields.byline
-
-    if (byline == null) throw new Error(`Byline was undefined in ${path}!`)
-
-    return [
-        internalid,
-        {
-            id: internalid,
-            path: path,
-            byline,
-            headline: title,
-            standfirst,
-            image: imageURL,
-            elements,
-        },
-    ]
+    if (byline == null) console.error(`Byline was undefined in ${path}!`)
+    
+    switch (result.type) {
+        case ContentType.ARTICLE: 
+            const parser = elementParser(path)
+
+            const image =
+                result &&
+                result.blocks &&
+                result.blocks.main &&
+                result.blocks.main.elements &&
+                result.blocks.main.elements[0].assets &&
+                extractImage(result.blocks.main.elements[0].assets)
+            const imageURL =
+                (image && image.file) ||
+                'https://media.guim.co.uk/d1c48b0c6ec594b396f786cfd3f6ba6ae0d93516/0_105_2754_1652/2754.jpg'
+        
+            const blocks =
+                result &&
+                result.blocks &&
+                result.blocks.body &&
+                result.blocks.body.map(_ => _.elements)
+            const body = blocks && blocks.reduce((acc, cur) => [...acc, ...cur], [])
+            if (body == null) throw new Error(`Body was undefined in ${path}!`)
+        
+            const elements = await attempt(Promise.all(body.map(parser)))
+            if (hasFailed(elements))
+                throw new Error(`Element parsing failed in ${path}!`) //This should not fire, the parser should log if anything async fails and then return the remainder.
+        
+            if (elements == null) throw new Error(`Elements was undefined in ${path}!`)
+        
+            return [
+                internalid,
+                <Article> {
+                    type: 'article',
+                    id: internalid,
+                    path: path,
+                    byline: byline || '',
+                    headline: title,
+                    standfirst: standfirst || '',
+                    image: imageURL,
+                    elements
+                },
+            ]
+
+        case ContentType.CROSSWORD: 
+
+            return [
+                internalid,
+                <CrosswordArticle> {
```

Comment:
This cast (and the one above) should be redundant just FYI, by no means a blocker.